src/network/mod.rs

```diff
@@ -266,12 +289,15 @@ fn connect_and_propagate_with<F>(
     let addr = match node.address() {
         Some(addr) => addr,
         None => {
-            info!("ignoring P2P node without an IP address: {:?}", node);
+            info!(
+                state.logger(),
+                "ignoring P2P node without an IP address: {:?}", node
+            );
             return;
         }
     };
     let node_id = node.id();
-    debug!("connecting to node {} at {}", node_id, addr);
+    debug!(state.logger(), "connecting to node {} at {}", node_id, addr);
```

Comment:
A `ConnectionState` is created two lines below, so maybe it should go under per-connection logger, and the "at {}" part of the message is no longer needed.